apps/firelens-stability/lib/cloud/ecs.ts

```diff
@@ -41,86 +46,229 @@ async function processEcsTestTaskQueue() {
+
+        /* Mark the ECS task as resolved. */
+        ecsTaskReturn(ecsTestTask);
+        return;
+    }
+    
+    /* Tasks need to be retried. Retry the ended tasks */
+    let taskArns = ecsTestTask.executionRecord.taskArns;
+    if (endedTasksCount !== 0) {
+        console.log(`  🔁 ${testCase.managed.collectionName}/${testCase.managed.suiteName}/${testCase.managed.caseName}: Retrying failed tasks`);
+        const taskCount = endedTasksCount;
+        let newTasksLaunched = await launchECSTasks(testCase, taskCount, ecs, ecsTestTask.taskDefinitionArn);
+        taskArns = [...startingTasks, ...runningTasks, ...newTasksLaunched];
+    }
+
+    /* Add back to queue for validation */
+    addECSTaskToQueue(10000, 20000, {
+        ...ecsTestTask,
+        executionRecord: {
+            ...ecsTestTask.executionRecord,
+            taskArns: taskArns, /* Replace the task arns with updated list */
+        },
+        executionRound: ecsTestTask.executionRound + 1,
+    });
+}
 
-        // Register task definition
-        let taskDefinitionArn: string;
+async function outOfRetriesECSTask(ecsTestTask: IEcsTestTask) {
+    const testCase = ecsTestTask.testCase;
+
+    // Set the region name
+    AWS.config.update({ region: ecsTestTask.testCase.config.region });
+
+    // Create an ECS client
+    const ecs = new AWS.ECS();
+
+    const {
+        runningTasksCount,
+    } = await validateECSTestTask(ecs, ecsTestTask);
+
+    /* Done! All tasks are in running */
+    if (runningTasksCount === ecsTestTask.testCase.config.taskCount) {
+        console.log(`  ✅ ${testCase.managed.collectionName}/${testCase.managed.suiteName}/${testCase.managed.caseName}: All tasks in running state`);
+
+        /* Mark the ECS task as resolved. */
+        ecsTaskReturn(ecsTestTask);
+        return;
+    }
+
+    /* Failed tasks */
+    else {
+        console.log(`  ❌ ${testCase.managed.collectionName}/${testCase.managed.suiteName}/${testCase.managed.caseName}: Some tasks failed to enter running state after ${Constants.ecsConfig.maxRetryRounds} rounds of retries failed.`);
+
+        /* Mark the ECS task as resolved. */
+        ecsTaskReturn(ecsTestTask);
+        return;
+    }
+}
+
+
+async function launchECSTasks(testCase: ITestCase, taskCount: number, ecs: AWS.ECS, taskDefinitionArn: string): Promise<string[]> {
+    let launchedTasks = [];
+    for (let i = 0; i < taskCount; i += Math.min(10, taskCount - i)) {
+        const count = Math.min(10, taskCount - i);
+        let result: PromiseResult<AWS.ECS.RunTaskResponse, AWS.AWSError>;
         try {
-            taskDefinitionArn = (await ecs.registerTaskDefinition(taskDefinition).promise()).taskDefinition!.taskDefinitionArn;
+            result = await ecs.runTask({
+            cluster: testCase.config.cluster,
+            taskDefinition: taskDefinitionArn!,
+            count: count,
+            launchType: "FARGATE",
```

Comment:
do we want to harcode the launchType??

Reply:
if we parametrize it from start its easier to change in future.

I think we will stick with Fargate long term tho, so its fine.